src/d3a/models/strategy/external_strategies/storage.py

```diff
@@ -176,11 +176,14 @@ def _delete_bid(self, payload):
 
     def _delete_bid_impl(self, arguments, response_channel):
         try:
-            self.remove_bid_from_pending(arguments["bid"], self.market.id)
-            self.state.offered_buy_kWh[self.market.time_slot] -= arguments["bid"]["energy"]
+            deleted_bids_dict = self.remove_bid_from_pending(self.market.id, arguments["bid"])
```

Comment:
I think this is not enough. You need to at least handle the lack of the `bid` argument, which will implicitly be treated as  'delete all bids'.

Reply:
The lack of the bid argument is handled in remove_bid_from_pending (case when bid is None).
Or do you mean something else?

Reply:
I meant that, if the endpoint receives a request with `arguments = {}` (without the bid argument at all), the endpoint should delete all bids. I know the function handles it, but only if we pass arguments["bid"] = None, not if we do not pass any argument at all (in this case it will crash with a `KeyError` in this line)

Reply:
Ahh, ok. Will handle this.

Reply:
Thanks!